pallets/nodle-staking/src/lib.rs

```diff
@@ -0,0 +1,2311 @@
+    {
+        pub fn new(validator: AccountId, amount: Balance) -> Self {
+            Nominator {
+                nominations: OrderedSet::from(vec![Bond {
+                    owner: validator,
+                    amount,
+                }]),
+                total: amount,
+                active_bond: amount,
+                unlocking: Vec::new(),
+            }
+        }
+        pub fn add_nomination(&mut self, bond: Bond<AccountId, Balance>) -> bool {
+            let amt = bond.amount;
+            if self.nominations.insert(bond) {
+                self.total += amt;
+                self.active_bond += amt;
+                true
+            } else {
+                false
+            }
+        }
+        // Returns Some(remaining balance), must be more than MinNominatorStake
+        // Returns None if nomination not found
+        pub fn rm_nomination(&mut self, validator: AccountId) -> Option<Balance> {
+            let mut amt: Option<Balance> = None;
+            let nominations = self
+                .nominations
+                .0
+                .iter()
+                .filter_map(|x| {
+                    if x.owner == validator {
+                        amt = Some(x.amount);
+                        None
+                    } else {
+                        Some(x.clone())
+                    }
+                })
+                .collect();
+            if let Some(balance) = amt {
+                self.nominations = OrderedSet::from(nominations);
+                // self.total -= balance;
+                self.active_bond -= balance;
+                Some(self.active_bond)
+            } else {
+                None
+            }
+        }
+        // Returns None if nomination not found
+        pub fn inc_nomination(&mut self, validator: AccountId, more: Balance) -> Option<Balance> {
+            for x in &mut self.nominations.0 {
+                if x.owner == validator {
+                    x.amount += more;
+                    self.total += more;
+                    self.active_bond += more;
+                    return Some(x.amount);
+                }
+            }
+            None
+        }
+        // Returns Some(Some(balance)) if successful
+        // None if nomination not found
+        // Some(None) if underflow
+        pub fn dec_nomination(
+            &mut self,
+            validator: AccountId,
+            less: Balance,
+        ) -> Option<Option<Balance>> {
+            for x in &mut self.nominations.0 {
+                if x.owner == validator {
+                    if x.amount > less {
+                        x.amount -= less;
+                        // self.total -= less;
+                        self.active_bond -= less;
+                        return Some(Some(x.amount));
+                    } else {
+                        // underflow error; should rm entire nomination if x.amount == validator
+                        return Some(None);
+                    }
```

Comment:
use safe math or saturating operators :)

Reply:
:+1:  Moved to saturating_sub()